### PR TITLE
Petrov day Opt In

### PR DESCRIPTION
**Pull request by Raemon**
┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1208395069275854) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/lib/collections/petrovDayActions/schema.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    optional: true,
+    canRead: ['guests'],
+    canCreate: ['members'],
+    canUpdate: ['members'],
+  },
+  userId: {
+    type: String,
+    canRead: ['guests'],
+    canCreate: ['members'],
+    canUpdate: ['members'],
```
Comment (b0b3rt) [inline packages/lesswrong/lib/collections/petrovDayActions/schema.ts:24]: `canUpdate` should probably be `admins` only, `canRead` should probably be `userOwns` and `admins` only.

File: packages/lesswrong/lib/collections/petrovDayActions/collection.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+export const PetrovDayActions: PetrovDayActionsCollection = createCollection({
+  collectionName: 'PetrovDayActions',
+  typeName: 'PetrovDayAction',
+  schema
+});
+
+addUniversalFields({collection: PetrovDayActions})
+
+export default PetrovDayActions;
```
Comment (b0b3rt) [inline packages/lesswrong/lib/collections/petrovDayActions/collection.ts:13]: We should probably have a `checkAccess` which forbids reads at the collection level unless you own the item (`userOwns`) or are an admin, in addition to gating reads at the field level.

File: packages/lesswrong/lib/collections/petrovDayActions/collection.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import schema from './schema';
+import { createCollection } from '../../vulcan-lib';
+import { addUniversalFields } from '../../collectionUtils'
+
+export const PetrovDayActions: PetrovDayActionsCollection = createCollection({
+  collectionName: 'PetrovDayActions',
+  typeName: 'PetrovDayAction',
+  schema
```
Comment (b0b3rt) [inline packages/lesswrong/lib/collections/petrovDayActions/collection.ts:8]: you'll need to add the default mutations.  `editCheck` shold return false.  `removeCheck` should return false.

Comment (b0b3rt) [reply packages/lesswrong/lib/collections/petrovDayActions/collection.ts:8]: `newCheck` will, in the future, need to run a query against previous actions taken by this user, to see if they have a specific role that allows them to take the action that they're trying to take with the new record

File: packages/lesswrong/components/common/LWHome.tsx [hunk 4]
```diff
@@ -26,8 +26,11 @@ const LWHome = () => {
             <FrontpageReviewWidget reviewYear={REVIEW_YEAR}/>
           </SingleColumnSection>}
           <SingleColumnSection>
-            <DismissibleSpotlightItem current/>
+            <OptIntoPetrovButton/>
           </SingleColumnSection>
+          {/* <SingleColumnSection>
+            <DismissibleSpotlightItem current/>
+          </SingleColumnSection> */}
```
Comment (b0b3rt) [inline packages/lesswrong/components/common/LWHome.tsx:33]: cleanup or leave a TODO to return it

File: packages/lesswrong/components/seasonal/PetrovDayWrapper.tsx [hunk 5]
```diff
@@ -21,13 +21,13 @@ const PetrovDayWrapper = () => {
   });
   
   
-  if (internalData?.PetrovDayCheckIfIncoming.launched) {
-    return <Components.PetrovDayLossScreen/>
-  } else {
-    return <Components.PetrovDayButton
+  // if (internalData?.PetrovDayCheckIfIncoming.launched) {
+    // return <Components.PetrovDayLossScreen/>
+  // } else {
```
Comment (b0b3rt) [inline packages/lesswrong/components/seasonal/PetrovDayWrapper.tsx:26]: cleanup or leave a TODO to return it

File: packages/lesswrong/components/seasonal/petrovDay/OptIntoPetrovButton.tsx [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+    void updateCurrentUser({
+      petrovPressedButtonDate: new Date()
+    });
+  }
+
+  const updateConfirmationCode = (event: React.ChangeEvent<HTMLInputElement | HTMLSelectElement | HTMLTextAreaElement>) => {
+    if (!alreadyLaunched) {
+      setConfirmationCode(event.target.value)
+    }
+  }
```
Comment (b0b3rt) [inline packages/lesswrong/components/seasonal/petrovDay/OptIntoPetrovButton.tsx:188]: Are these going to be used?

**Review by b0b3rt — APPROVED** [review]

File: packages/lesswrong/components/seasonal/petrovDay/OptIntoPetrovButton.tsx (hunk 6 above)
Comment (Raemon) [reply packages/lesswrong/components/seasonal/petrovDay/OptIntoPetrovButton.tsx:199]: those are used!

File: packages/lesswrong/lib/collections/petrovDayActions/collection.ts [hunk 7]
```diff
@@ -0,0 +1,31 @@
+import schema from './schema';
+import { createCollection } from '../../vulcan-lib';
+import { addUniversalFields, getDefaultResolvers } from '../../collectionUtils'
+import { getDefaultMutations, MutationOptions } from '../../vulcan-core/default_mutations';
+
+const options: MutationOptions<DbPetrovDayAction> = {
+  newCheck: (user: DbUser|null) => {
+    return true
```
Comment (b0b3rt) [inline packages/lesswrong/lib/collections/petrovDayActions/collection.ts:8]: this can be omitted if it's just returning true

**Review by b0b3rt — APPROVED** [review]

**Pull request merged**